extensions/attributegen/config.proto

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2020 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package istio.attributegen;
+
+// Configuration to generate attributes based on existing attributes.
+message PluginConfig {
+  // next id: 2
+  // Multiple independent attribute generation configurations.
+  repeated AttributeGeneration attributes = 1;
+}
+
+// AttributeGeneration uses [well known attributes]
+// (https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/rbac_filter#condition)
+// as inputs and produces a new attribute.
+// The new attribute can be used by a downstream plugin.
+//
+// Example: AttributeGen may produce an attribute called `istio.operationId` which is analogous to
+// the `operationId` in the OpenAPI Spec.
+// The Stats plugin will run after this plugin and use `istio.operationId` to populate a dimension.
+//
+// If multiple AttributeGenerations produce the same attribute, the result of the last generation
```

Comment:
Using last is fine, but I recall that in Mixer once an attribute had a value, any subsequent updates were ignored.